### PR TITLE
dependency: Updated Maven version to 3.6.3

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Resolves: #13970

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1329,7 +1329,7 @@
                   <version>${java.version}</version>
                 </requireJavaVersion>
                 <requireMavenVersion>
-                  <version>3.3.9</version>
+                  <version>3.8.8</version>
```
[the remark this answers is not in the document]
Comment (relentless-pursuit) [reply pom.xml:1332]: Okay. So, should the PR be referenced to Maven 3.8.6? I mean should i make changes to other files where i have mentioned 3.8.8 and replace it with 3.8.6?

[the remark this answers is not in the document]
Comment (romani) [reply pom.xml]: Pom should use 3.8.6, all CI can use any above version.

File: pom.xml [hunk 2]
```diff
@@ -1329,7 +1329,7 @@
                   <version>${java.version}</version>
                 </requireJavaVersion>
                 <requireMavenVersion>
-                  <version>3.3.9</version>
+                  <version>3.8.6</version>
```
Comment (romani) [inline pom.xml:1332]: I updated issue title to make it more clear
https://github.com/checkstyle/checkstyle/issues/13970

**Review by romani — CHANGES_REQUESTED** [review]
Please use reference to this issue

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (relentless-pursuit) [reply pom.xml]: > Pom should use 3.8.6, all CI can use any above version.

Shouldn't POM be 3.5.4 as the description now says minimum Maven version be 3.5.4. Please correct me if i have understood it wrong. And, all CIs i guess are having more of less 3.8.8 and hence no need to change them.

[the remark this answers is not in the document]
Comment (romani) [reply pom.xml]: >The plugin org.codehaus.mojo:xml-maven-plugin:1.1.0 requires Maven version 3.5.4

Yes, 3.5.4.

**Comment by romani** [comment]
Please use #13970 as commit prefix.

**Comment by relentless-pursuit** [comment]
> Please use #13970 as commit prefix.

done

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (relentless-pursuit) [reply pom.xml]: done

**Comment by relentless-pursuit** [comment]
> From [#13887 (comment)](https://github.com/checkstyle/checkstyle/pull/13887#issuecomment-1763414859)
> 
> > Sad that non of CI jobs user a version that we declare in pom to report problem to us automatically
> 
> From [#14097 (comment)](https://github.com/checkstyle/checkstyle/pull/14097#discussion_r1414664615)
> 
> > Yes, 3.5.4.
> > Only this version only in pom.
> > CIs completely ok to use bigger versions.
> 
> @romani seems like we have conflict between why issue was created, and sentiment in PR. Do we have CI job that runs `mvn clean verify` on same version that is in the pom?

@nrmancuso, why not use a relatively higher maven version like 3.8.8 in pom.xml. I understand that chosing the minimal version stable release is standard practice. But, what could be the unintended consequences if we go with a relatively newer version?

**Comment by nrmancuso** [comment]
>@nrmancuso, why not use a relatively higher maven version like 3.8.8 in pom.xml. I understand that chosing the minimal version stable release is standard practice. But, what could be the unintended consequences if we go with a relatively newer version?

This is debatable for sure, but I think the idea is to allow contributors and users to not have to update their maven version.  Let's get clarity on https://github.com/checkstyle/checkstyle/pull/14097#issuecomment-1840783976 before we proceed.

**Comment by romani** [comment]
> Do we have CI job that runs mvn clean verify on same version that is in the pom?

I do not think so. Ideally we should use min required maven in all CIs, to easily catch fact that some plugin requires bigger version. But during container creation, you got strong opinion that we need newer version, I thought on this position for a bit, and here is nuance. We spread different types of validation for different CIs, not all CI can be easily forced to have certain version of maven. I might be easier for us to rely on users feedback that our min required version is outdated. Reasoning is new contributor will download latest maven, old contributor already advanced enough to fix such mismatch easily.

**Comment by nrmancuso** [comment]
>We spread different types of validation for different CIs, not all CI can be easily forced to have certain version of maven.

I agree that it is good to have a few different versions in CI. Docker image we created is only for no-exception/no-error, it was better to have newer version to not have to update image as soon. For `mvn clean verify` execution, we should have at least one execution on actual pom.xml version.

**Comment by romani** [comment]
@relentless-pursuit , please update  cirrus.yml to 3.5.4

**Review by romani — APPROVED** [review]
Ok to merge if CI pass

**Comment by nrmancuso** [comment]
@relentless-pursuit I am prepared to approve and merge this once CI is green, let's get this across the finish line 🚀

**Comment by relentless-pursuit** [comment]
> Bingo!
> 
> From CI:
> 
> > Unable to parse command line options: Unrecognized option: --no-transfer-progress
> 
> Let's find when this parameter was added and let's bump minimum version to it.
> 
> https://stackoverflow.com/a/21638827/1015848 3.6.1

Thank you @Romani for the reference.

File: .cirrus.yml [hunk 3]
```diff
@@ -29,9 +29,10 @@ task:
     MAVEN_OPTS: "-Dhttp.keepAlive=false \
       -Dmaven.repo.local=%CIRRUS_WORKING_DIR%/.m2 \
       -Dmaven.wagon.http.retryHandler.count=3"
-    MAVEN_VERSION: "3.8.4"
+    # This Maven version here should be same as minimum defined maven version in pom.xml
```
Comment (nrmancuso) [inline .cirrus.yml:32]: @relentless-pursuit attention to your own comment :)

Comment (relentless-pursuit) [reply .cirrus.yml:32]: Thanks for pointing it out. The issue says minimum version be 3.5.4 and the version which possibly handles the CI failure is 3.6.1. Shall i remove the comment?

Comment (nrmancuso) [reply .cirrus.yml]: >Let's find when this parameter was added and let's bump minimum version to it.

@romani to be clear: you are good to update project minimum version to 3.6.1 (pom.xml, documentation)? This version is almost five years old at this point: https://maven.apache.org/docs/history.html

Comment (relentless-pursuit) [reply .cirrus.yml]: > > Let's find when this parameter was added and let's bump minimum version to it.
> 
> @romani to be clear: you are good to update project minimum version to 3.6.1 (pom.xml, documentation)? This version is almost five years old at this point: https://maven.apache.org/docs/history.html, it should be fine.

the CI was fialing with 3.6.1:
```
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:build-helper-maven-plugin:3.5.0:add-source (add-source) on project checkstyle: The plugin org.codehaus.mojo:build-helper-maven-plugin:3.5.0 requires Maven version 3.6.3 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:build-helper-maven-plugin:3.5.0:add-source (add-source) on project checkstyle: The plugin org.codehaus.mojo:build-helper-maven-plugin:3.5.0 requires Maven version 3.6.3
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:172)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginIncompatibleException: The plugin org.codehaus.mojo:build-helper-maven-plugin:3.5.0 requires Maven version 3.6.3
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.checkRequiredMavenVersion (DefaultMavenPluginManager.java:309)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:168)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginIncompatibleException
validation failed with error code #1.
```

Comment (relentless-pursuit) [reply .cirrus.yml]: `build-helper-maven-plugin` version `3.5.0` requires minimum version to be `3.6.3`

Comment (romani) [reply .cirrus.yml]: Min version in pom should be any version that shown at error like `Caused by: org.apache.maven.plugin.PluginIncompatibleException: The plugin org.codehaus.mojo:build-helper-maven-plugin:3.5.0 requires Maven version 3.6.3`

To fail build quickly and be resolved quickly by user.

I never think on such version, if some plugin complains, we bump our min version up up to min required by plugins we use .

Comment (romani) [reply .cirrus.yml]: Please use `3.6.3`.

**Comment by romani** [comment]
@relentless-pursuit , cirrus is not happy

**Review by romani — APPROVED** [review]
Ok to merge.

Thanks a lot!!

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Please update commit message and PR title to 3.6.3

**Comment by relentless-pursuit** [comment]
> Please update commit message and PR title to 3.6.3

done

**Review by nrmancuso — APPROVED** [review]
Awesome, @relentless-pursuit thanks a lot for this!

**Pull request merged**